include/drivers/clock_control/stm32_clock_control.h

```diff
@@ -164,12 +164,29 @@
 #define STM32_CSI_FREQ		0
 #endif
 
-#if DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(clk_lsi), fixed_clock, okay)
+#if defined(CONFIG_SOC_SERIES_STM32WBX)
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_watchdog)
+/* IWDG automatically enables the LSI clock : get its freq. */
+#define STM32_LSI_FREQ		DT_PROP(DT_NODELABEL(clk_lsi1), clock_frequency)
+#elif DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(clk_lsi1), fixed_clock, okay)
+#define STM32_LSI_ENABLED	1
+#define STM32_LSI_FREQ		DT_PROP(DT_NODELABEL(clk_lsi1), clock_frequency)
+#else
+/* LSI clock freq should be in device tree, not from stm32Cube LSI_VALUE */
+#define STM32_LSI_FREQ		0
+#endif
+#else /* CONFIG_SOC_SERIES_STM32WBX */
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_watchdog)
+/* IWDG automatically enables the LSI clock : get its freq. */
+#define STM32_LSI_FREQ		DT_PROP(DT_NODELABEL(clk_lsi), clock_frequency)
+#elif DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(clk_lsi), fixed_clock, okay)
 #define STM32_LSI_ENABLED	1
 #define STM32_LSI_FREQ		DT_PROP(DT_NODELABEL(clk_lsi), clock_frequency)
 #else
+/* LSI clock freq should be in device tree, not from stm32Cube LSI_VALUE */
 #define STM32_LSI_FREQ		0
 #endif
+#endif /* CONFIG_SOC_SERIES_STM32WBX */
```

Comment:
Please review this section.
1- LSI is common to the whole system, so is this file. We should not see driver specific code in this file (`st_stm32_watchdog` in this case).
2- If series specific code is required, please use compatilble to catch with what we've done till now in this file (no `CONFIG_SOC_SERIES_STM32WBX`)

Reply:
1. The iwdt is a special case where the frequency of the LSI (or LSI1) RC oscillator is needed from the **clk-lsi node** of the soc DTS even if this node is not enabled in the board.
The STM32_LSI_FREQ is exactly what is needed by the iwdg to calculate its timeout value. However we could still define a STM32_LSI_CLOCK for the iwdg purpose only.

2. what about devices like stm32wb with LSI 1 (and LSI2) RC oscillator instead of LSI : the STM32_LSI_FREQ is the LSI1 RC oscillator. Should the STM32_LSI_FREQ be defined as well

Reply:
using the LSI clock for iwdt purpose could be :
```
/* the STM32_LSI_FREQ is given by the LSI or LSI1 RC oscillator */
#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_watchdog)
/* when IWDG is on, the LSI is not necessarily enabled, just get its freq. */
#if DT_PROP(DT_NODELABEL(clk_lsi), clock_frequency)
#define STM32_LSI_FREQ		DT_PROP(DT_NODELABEL(clk_lsi), clock_frequency)
#elif DT_PROP(DT_NODELABEL(clk_lsi1), clock_frequency)
#define STM32_LSI_FREQ		DT_PROP(DT_NODELABEL(clk_lsi1), clock_frequency)
#else
/* LSI clock freq should be in device tree, not from stm32Cube LSI_VALUE */
#define STM32_LSI_FREQ		0
#endif
#else /* st_stm32_watchdog */
 #if DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(clk_lsi), fixed_clock, okay)
 #define STM32_LSI_ENABLED	1
 #define STM32_LSI_FREQ		DT_PROP(DT_NODELABEL(clk_lsi), clock_frequency)
#elif DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(clk_lsi1), fixed_clock, okay)
#define STM32_LSI_ENABLED	1
#define STM32_LSI_FREQ		DT_PROP(DT_NODELABEL(clk_lsi1), clock_frequency)
 #else
 #define STM32_LSI_FREQ		0
 #endif
#endif /* st_stm32_watchdog */
 
 #if DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(clk_hsi), st_stm32h7_hsi_clock, okay)
 #define STM32_HSI_ENABLED	1
```